types/ramda/index.d.ts

```diff
@@ -1447,9 +1447,8 @@ declare namespace R {
 
         /**
          * Returns a function that when supplied an object returns the indicated property of that object, if it exists.
-         * Note: TS1.9 # replace any by dictionary
          */
-        prop<P extends string, T>(p: P, obj: Record<P, T>): T;
+        prop<P extends keyof T, T>(p: P, obj: T): T[P];
         prop<P extends string>(p: P): <T>(obj: Record<P, T>) => T;
```

Comment:
Please update the curried version as well

Reply:
@samsonkeung I don't know why, but curried version seems to work.

Reply:
If I wrote it like `prop<P extends keyof T, T>(p: P): (obj: T) => T[P];`, it wouldn't (I don't know how to write it in different way so that it would work as expected. (It works as expected now though).